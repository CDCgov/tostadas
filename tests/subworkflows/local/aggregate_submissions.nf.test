nextflow_workflow {

    name "Test Workflow AGGREGATE_SUBMISSIONS"
    script "subworkflows/local/aggregate_submissions.nf"
    workflow "AGGREGATE_SUBMISSIONS"

    test("Should aggregate submission reports and merge accessions") {

        when {
            params {
                submission_prod_or_test = "test"
                dry_run = true
                metadata_basename = "rsv_test_metadata"
                submission_mode = "ftp"
                env_yml = "${projectDir}/environment.yml"
            }

            workflow {
                """
                input[0] = [ [ batch_id: 'batch_1' ], file("${projectDir}/tests/modules/local/fetch_reports/batch_1/") ]
                input[1] = file("${projectDir}/conf/submission_config.yaml")
                input[2] = file("${projectDir}/tests/subworkflows/local/validated_metadata_all_samples.tsv")
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
