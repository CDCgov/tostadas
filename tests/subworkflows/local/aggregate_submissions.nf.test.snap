{
    "Should aggregate submission reports and merge accessions": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "accession_augmented_xlsx": [
                    
                ],
                "all_report_csvs": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-28T16:25:35.183361848"
    }
}